src/graphql/utils/loaderFunction.ts

```diff
@@ -0,0 +1,26 @@
+import fetchWithAbort from './fetchWithAbort'
+
+// A function returns a Promise that resolves to the values corresponding the keys
+function loaderFunction(keys: readonly string[], token = ''): Promise<Array<Promise<Object>>> {
+  // Return a Promise that resolves to the values corresponding the keys
+  return new Promise(async resolve => {
```

Comment:
Do we need to construct this Promise? Isn't it possible to return the `Promise.all` call directly?

Reply:
Nope, `dataloader` expects a single Promise

Reply:
Doesn't `Promise.all` return a single Promise that resolves with all the values though? I believe we could write the code as follows and still get the same result:

```ts
import fetchWithAbort, { ErrorResult } from './fetchWithAbort'

function loaderFunction(keys: readonly string[], token = ''): Promise<Array<Object | ErrorResult>> {
  return Promise.all(
    keys.map((key: string) =>
      fetchWithAbort(
        key,
        token && token.length > 0
          ? {
              headers: { authorization: token },
            }
          : {},
      ),
    ),
  )
}

export default loaderFunction
```

Or am I missing something obvious?